src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java

```diff
@@ -107,25 +110,30 @@ public ResponseEntity<PreContingencyResult> getNResult(@Parameter(description =
         @ApiResponse(responseCode = "404", description = "Security analysis result has not been found")})
-    public ResponseEntity<List<ContingencyResultDTO>> getNmKContingenciesResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid) {
-        List<ContingencyResultDTO> result = service.getNmKContingenciesResult(resultUuid);
+    public ResponseEntity<Page<ContingencyResultDTO>> getNmKContingenciesResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid,
+                                                                                    @Parameter(description = "Filters") @RequestParam(name = "filters", required = false) String stringFilters,
+                                                                                    @Parameter(description = "Pagination parameters") Pageable pageable) throws JsonProcessingException {
+        List<ResourceFilterDTO> filters = ResourceFilterDTO.fromStringToList(stringFilters);
+        Page<ContingencyResultDTO> result = service.getNmKContingenciesResult(resultUuid, filters, pageable);
 
         return result != null
             ? ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(result)
             : ResponseEntity.notFound().build();
     }
 
-    @GetMapping(value = "/results/{resultUuid}/nmk-constraints-result", produces = APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/results/{resultUuid}/nmk-constraints-result/paged", produces = APPLICATION_JSON_VALUE)
     @Operation(summary = "Get a security analysis result from the database - NMK contingencies result")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The security analysis result"),
         @ApiResponse(responseCode = "404", description = "Security analysis result has not been found")})
-    public ResponseEntity<List<SubjectLimitViolationResultDTO>> getNmKConstraintsResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid) {
-
-        List<SubjectLimitViolationResultDTO> result = service.getNmKConstraintsResult(resultUuid);
+    public ResponseEntity<Page<SubjectLimitViolationResultDTO>> getNmKConstraintsResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid,
+                                                                                        @Parameter(description = "Filters") @RequestParam(name = "filters", required = false) String stringFilters,
+                                                                                        @Parameter(description = "Pagination parameters") Pageable pageable) throws JsonProcessingException {
+        List<ResourceFilterDTO> filters = ResourceFilterDTO.fromStringToList(stringFilters);
+        Page<SubjectLimitViolationResultDTO> result = service.getNmKConstraintsResult(resultUuid, filters, pageable);
```

Comment:
Use the controller to decode your list and let the service deserialize into your DTOs with Jackson.
```suggestion
                                                                                        @Parameter(description = "Filters") @RequestParam(name = "filters", defaultValue = "", required = false) List<String> stringFilters,
                                                                                        @Parameter(description = "Pagination parameters") Pageable pageable) throws JsonProcessingException {
        Page<SubjectLimitViolationResultDTO> result = service.getNmKConstraintsResult(resultUuid, stringFilters, pageable);
```

Reply:
fixed